src/Router.py

```diff
@@ -0,0 +1,158 @@
+
+    def handle_arp(self, arp_packet, ethernet_detailed, ethertype, interface):
+        '''
+        Handles ARP requests by unpacking them and constructing a respose packet (if needded).
+        Prints the ARP packet contents.
+        '''
+        arp_header = arp_packet[0][14:42]       # ARP is 28 bytes payload
+        arp_detailed = struct.unpack(Router.ARP_FORMAT, arp_header)
+        self.print_arp(arp_detailed, ethernet_detailed, ethertype)
+        
+        if self.common_network(socket.inet_ntoa(arp_detailed[8]), socket.inet_ntoa(arp_detailed[6])):
+            return None
+        
+        opcode = bytes.fromhex('0002')
+        
+        if interface == 'net1':
+            source_mac = bytes.fromhex(Router.MAC_ROUTER_NET1)
+            source_ip = bytes.fromhex('01010101')
+        else:
+            source_mac = bytes.fromhex(Router.MAC_ROUTER_NET2)
+            source_ip = bytes.fromhex('02020201')
+
+        new_ethernet_header = ethernet_detailed[1] + source_mac + ethertype
+        arp_response = arp_header[:6] + opcode + source_mac + source_ip + arp_detailed[5] + arp_detailed[6]
+        response_packet = new_ethernet_header + arp_response
+        return response_packet
+
+    def handle_ip(self, ip_packet, ethertype, ethernet_detailed):
+        ip_header = ip_packet[0][14:34]       # IP is 20 bytes payload
+        ip_detailed = struct.unpack("9s1s2s4s4s", ip_header)
+        icmp_header = ip_packet[0][34:42]       # ICMP is 8 bytes payload
+        icmp_detailed = struct.unpack("1s1s2s4s", icmp_header)
+        self.print_ping(ethertype, ethernet_detailed, ip_detailed, icmp_detailed)
+        
+        if icmp_detailed[0] == b'08':
+            new_ethernet_header = bytes.fromhex(Router.CLIENT1_MAC) + bytes.fromhex(Router.MAC_ROUTER_NET1) + ethertype
+            ping_response = new_ethernet_header + ip_header + icmp_header 
+            return ping_response
+        elif icmp_detailed[0] == b'00':
+            new_ethernet_header = bytes.fromhex(Router.CLIENT2_MAC) + bytes.fromhex(Router.MAC_ROUTER_NET2) + ethertype
+            ping_response = new_ethernet_header + ip_header + icmp_header 
+            return ping_response
+        else: 
+            return None
+
+    def create_ip_packet(self, request_packet):
+        return request_packet
+
+    def observe_traffic(self):
+        '''
+        Gets incoming packets from net1 net2 interfaces and handles them.
+        '''
+        with socket.socket(socket.AF_PACKET, socket.SOCK_RAW, socket.ntohs(0x0003)) as rawSocket:
+            while True:
+                packet = rawSocket.recvfrom(4096)
+                (_, interface) = packet
+                if (interface[0] != 'net1') and (interface[0] != 'net2'):
+                    continue
+                # ethernet_unpacking
+                interface = interface[0]
+                ethernet_header = packet[0][0:14]       # ARP is 14 bytes payload
```

Comment:
If you are referencing packet[0], why did you discard it when you unpacked packet?